duckduckgo/Android
# Sync: Add field validation

Task/Issue URL: https://app.asana.com/0/488551667048375/1206754257727806/f 

### Description
When calculating diff folder changes, for missing entities in the local db, we should stop adding them as deleted from folder. Instead just keep data as it is.

### Steps to test this PR

Best way to test this PR without hardcoding lines is by using `feature/cristian/sync/filter_invalid_items_patch` branch. That way you could test what happens when entities are missing in the payload.

_Feature 1_
- [ ] Checkout branch `feature/cristian/sync/filter_invalid_items_patch`
- [ ] fresh install in 2 devices
- [ ] go to Device A
- [ ] create a sync account
- [ ] go to device B
- [ ] join the sync account
- [ ] go back to Device A
- [ ] turn on airplane mode (we are going to create invalid bookmarks)
- [ ] create a bookmark (Bookmark1A), and edit title to have >3000 chars
- [ ] turn off airplane
- [ ] make something to trigger sync (e.g: access bookmarks screen)
- [ ] Take note of Bookmark1A Id (we will need it later)
- [ ] go back to Device B
- [ ] add this filter in your logcat for device B `package:mine message~:"Sync-Bookmarks-Metadata"`
(now we will perform some actions on DeviceB, ensure you **NEVER** see the Bookmark1A Id in the bookmarks_root Diff as Deleted)
- [ ] go to bookmarks screen, that should trigger sync
- [ ] add a bookmark (bookmark1B)
- [ ] remove the bookmark1B

_Ensure relations are not removed_
- [ ] create some bookmarks and folders
- [ ] From the IDE, go to inspect Database -> saved sites  -> entity table
- [ ] choose a random bookmark and take note of the bookmark id
- [ ] change that bookmark id manually to something different from the inspect database view
- [ ] In the UI that bookmark will disappear, however relations to the old id should not disappear
- [ ] move/sort bookmarks
- [ ] ensure we never lose the relation to the old Id
- [ ] (optional) again in the db, update the bookmark id again with the right Id
- [ ] (optional) now that entity should be visible again in the UI

### UI changes
| Before  | After |
| ------ | ----- |
!(Upload before screenshot)|(Upload after screenshot)|
